File: AssassinsGame/Player.py
#! /usr/bin/python3
class Player:
	def __init__(self,name,email,pin,kills,status):
		self.status = status
		self.name = name
		self.email= email
		self.pin = pin
		self.kills = kills
	def addAssassins(killer,target):
		self.killer = killer
		self.target = target
	def kill(pin,self,target):
		if self.status == true:
			target.die(pin,self)
			print ("killed")

	def die(pin,killer):
		if pin == self.pin and self.killer == killer:
			self.status = False
			return True	
	

print ("Hellow")
p1 = Player(self,"calvin","tacalvin@yahoo.com","1",0,True)

p2 = Player(self,"bill","tacalvin@yahoo.com","2",0,True)
p3 = Player(self,"joe","tacalvin@yahoo.com","3",0,True)
p1.addAssassins(p2,p3)
p2.addAssassins(p3,p1)
p3.addAssassins(p1,p2)
	
p1.kill(p3.pin,self,p3)

File: AssassinsGame/Assassin/Player.cpp
#include "Player.h"



Player::Player(string name, /*string email,*/ bool status, int kills, int pin): /*email(""), */status(1), kills(0)
{
	(*this).name = name;
	//(*this).email = email;
	(*this).status = status;
	(*this).kills = kills;
	(*this).pin = pin;
}

string Player::getName()
{
	return name;
}

bool Player::getStatus()
{
	return status;
}

int Player::getPin()
{
	return pin;
}

int Player::getKills()
{
	return kills;
}

Player * Player::getTarget()
{
	return target;
}

Player * Player::getKiller()
{
	return killer;
}

void Player::addAssassins(Player* killer, Player* target)
{
	
	(*this).target = target;
	(*this).killer = killer;
}

Player* Player::die(int playerpin, Player* actingkiller)
{
	if (pin == playerpin && (*this).getKiller() == actingkiller)
	{
		(*this).status = false;
		
		cout << "Player Killed" << endl;
		return (*this).getTarget();
	}

	else
		cout << "error invalid kill" << endl;
}



void Player::kill(int pin, Player* target)
{	
	
	if ((*this).getStatus())
	{
		cout << "Checking kill" << endl;
		(*target).die(pin, this);
		
	}
}



